### ISSUE TITLE
[TableView] Allow focus to set on both header and the subsequent dialog action button with keyboard

### ISSUE THREAD

**Issue by fhschan** [adobe/react-spectrum#2519]
# 🙋 Feature Request
Allow keyboard navigation to focus on table view column as well as a dialog action button if there is one in the column header.

## 🤔 Expected Behavior
- When using keyboard navigation, user can first focus on the column header before focus next sets to the dialog action button within the column header.
- User can still select the column header if the table allows column sorting
- Voice over announcement on the column header still work

## 😯 Current Behavior
Using keyboard navigation, the focus jumps directly to the column's action button if there is one, and user does not have a chance to focus on the column header itself.

## 🔦 Context
For tables, there could be some use cases where where we want to include a dialog trigger with an action button to display a small dialog for more information about the column. 

Currently with this behavior, the keyboard focus sets on the action button, but there is no way for user to use keyboard to focus on the column header itself.

Where when you tab into focus on the table, you can see when you use arrow keys on the table column, the focus immediately sets to the dialog's action button instead of the column header.

I've created a sandbox in the example section below which can hopefully give a better illustration on the request.

## 💻 Examples

I have a sample code here for reference:
https://codesandbox.io/s/zen-fermi-zbo48?file=/src/App.js

![Nov-17-2021 12-03-17](https://user-images.githubusercontent.com/22137751/142274056-8f0aa2d5-4bde-41a8-8492-6750c1570f8b.gif)

## 🧢 Your Company/Team
Adobe AEP

**Comment by matthewdeutsch** [comment]
Can't have nested interactive elements, as the whole column header is clickable to sort. Closing out issue - will work with team to find alternative place for column header info
